tests/toolbox_image.sh

```diff
@@ -81,6 +147,9 @@ failure_exit()
     echo "  sudo sh -c '\\"
     echo "    umount '${rootfs_dir}/${ARM_EMU_BIN}' && \\"
     echo "    unlink '${rootfs_dir}/${ARM_EMU_BIN}' && \\"
+    echo "    for loop_device in /dev/loop[0-9]; do ( if grep '${loop_device}' '/proc/mounts'; \\ "
+    echo "    then sudo umount '${rootfs_dir}${loop_device}'; fi ); done && \\"
+    echo "    for loop_device in /dev/loop[0-9]; do unlink '${rootfs_dir}${loop_device}'; done && \\"
```

Comment:
why two loops?
```sh
    echo "    for loop_device in /dev/loop*; do \\"
    echo "        if grep '\${loop_device}' '/proc/mounts'; then \\"
    echo "            umount '${rootfs_dir}\${loop_device}'; \\"
    echo "        fi; \\"
    echo "        if [ -f '${rootfs_dir}\${loop_device}' ]; then \\"
    echo "            unlink '${rootfs_dir}\${loop_device}' ]; \\"
    echo "        fi; \\"
    echo "  done && \\"
```
you want to escape the $ on loop_device, to prevent it from resolving, but ${rootfs_dir} should be resolved.

more importantly however:
loop[0-9] won't umount loop10
I am quite unhappy with the /dev/loop* however as it will umount ALL loop devices, INCLUDING the ones I have unrelated to this test!
I think what would be better, is to get all loop mount devices from /proc/mount that match rootfs_dir.
something like
```sh
loop_devices="$(grep '"${rootfs_dir}"*/dev/loop' /proc/mounts)"
```
Not sure if the * works without special quotes or anything, and not sure if you can do this with busybox grep. I think you need grep -E (to do regex) and then the asterix becomes .* (e.g. match any character inbetween) (if the order is even right). But that will atleast all valid devices!

see https://www.thegeekstuff.com/2011/10/grep-or-and-not-operators for reference

Reply:
Yes. 
The code is changed, the scenario where failrue exit is printed and the loop devices are mounted is not possible anymore, therefore removed the this code.